src/cmd/list.go

```diff
@@ -187,10 +187,15 @@ func listOutput(images, containers []map[string]interface{}) {
 			nameKey = "names"
 			createdKey = "created"
 		}
+
+		if podman.CheckVersion("2.0.0") {
+			idKey = "Id"
+		}
+
 		for _, image := range images {
 			id := utils.ShortID(image[idKey].(string))
-			name := image[nameKey].([]interface{})[0].(string)
-			created := image[createdKey].(string)
+			name := image[nameKey].([]interface{})[0]
```

Comment:
After pondering over this a bit, I don't think we should drop the type cast just like that.

I understand that the strict cast makes the code more sensitive to (future) changes in the JSON, and currently it doesn't matter because everything is a `string` anyway. However, we should also be in control of what we are printing, and we shouldn't just throw some unknown type at the `%s` format specifier in the hope that it will somehow work out.

Go does offer some nice features like [type assertions](https://tour.golang.org/methods/15) and [type switches](https://tour.golang.org/methods/16). These could be used to come up with a more elegant solution that's more tolerant towards JSON changes, and at the same time lets us retain control over our types.

At the very least, if something is not of the expected type, we should log some debug message with details about the actual type. Parsing the JSON and dealing with all the `interface` types can be tricky at times, so logging the actual failure can be useful.

As for the actual output, we could either print an empty string for all values that aren't of the expected type, or we can skip the entire column altogether. Or maybe something else. It somewhat depends on how practical each of these options are.

Either way, we should use a separate pull request for this topic because it's not directly connected to the issue of *JSON changes in Podman 2.0*. :)

Reply:
Sure! This just looked to me like the easiest solution if something were to change and we weren't fast enough to change it. Using type assertions/switches definitely gives us more power over dealing with what to show/hide in case of an error.

I'll look into that in the future.